uplink/metainfo/kvmetainfo/paths.go

```diff
@@ -8,21 +8,20 @@ package kvmetainfo
 //   since the exact previous key is
 //     append(previousPrefix(cursor), infinite(0xFF)...)
 
-// TODO commented until we will decide if we will support direction for objects listing
-// func keyBefore(cursor string) string {
-// 	if cursor == "" {
-// 		return ""
-// 	}
+func keyBefore(cursor string) string {
+	if cursor == "" {
+		return ""
+	}
 
-// 	before := []byte(cursor)
-// 	if before[len(before)-1] == 0 {
-// 		return string(before[:len(before)-1])
-// 	}
-// 	before[len(before)-1]--
+	before := []byte(cursor)
+	if before[len(before)-1] == 0 {
+		return string(before[:len(before)-1])
+	}
+	before[len(before)-1]--
 
-// 	before = append(before, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f)
-// 	return string(before)
-// }
+	before = append(before, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f)
```

Comment:
Let's use the approach described in https://github.com/storj/storj/blob/master/docs/blueprints/path-component-encoding.md

Reply:
hmm it looks that current code is appending to unencoded cursor but according to doc `Proceeding` should be constructed from encoded value? Am I right?

Reply:
Yes, it should happen with encoded values.